DNN Platform/Library/Entities/Portals/PortalSettingsService.cs

```diff
@@ -0,0 +1,216 @@
+            where T : IConvertible
+        {
+            try
+            {
+                var decryptedText = FIPSCompliant.DecryptAES(this[key], passPhrase, Host.GUID);
+                return (T)Convert.ChangeType(decryptedText, typeof(T));
+            }
+            catch (Exception exception)
+            {
+                Logger.Error(exception);
+            }
+
+            return default(T);
+        }
+
+        /// <inheritdoc />
+        public string GetEncrypted(string key) =>
+            this.GetEncrypted<string>(key);
+
+        /// <inheritdoc />
+        public string GetEncrypted(string key, string passPhrase) =>
+            this.GetEncrypted<string>(key, passPhrase);
+
+        /// <inheritdoc />
+        public void Delete(string key) =>
+            this.DeleteService.Delete(key, true);
+
+        /// <inheritdoc />
+        public void DeleteAll() =>
+            this.DeleteService.DeleteAll(true);
+
+        /// <inheritdoc />
+        public void DeleteAll(bool clearCache)
+        {
+            this.DeleteService.DeleteAll(clearCache);
+
+            if (clearCache)
+            {
+                this.Settings = this.LoadSettingsCallback();
+            }
+        }
+
+        /// <inheritdoc />
+        public void Delete(string key, bool clearCache)
+        {
+            this.DeleteService.Delete(key, clearCache);
+
+            if (clearCache)
+            {
+                this.Settings = this.LoadSettingsCallback();
+            }
+        }
+
+        /// <inheritdoc />
+        public void Update(string key, string value) => 
+            this.SaveService.Update(key, value, false);
+
+        /// <inheritdoc />
+        public void UpdateEncrypted(string key, string value) => 
+            this.SaveService.UpdateEncrypted(key, value, false);
+
+        /// <inheritdoc />
+        public void UpdateEncrypted(string key, string value, string passPhrase) =>
+            this.SaveService.UpdateEncrypted(key, value, passPhrase, false);
+
+        /// <inheritdoc />
+        public void Update(string key, string value, bool clearCache)
+        {
+            this.SaveService.Update(key, value, clearCache);
+
+            if (clearCache)
+            {
+                this.Settings = this.LoadSettingsCallback();
+            }
+        }
+
+        /// <inheritdoc />
+        public void UpdateEncrypted(string key, string value, bool clearCache)
+        {
+            this.SaveService.UpdateEncrypted(key, value, clearCache);
+
+            if (clearCache)
+            {
+                this.Settings = this.LoadSettingsCallback();
+            }
+        }
+
+        /// <inheritdoc />
+        public void UpdateEncrypted(string key, string value, string passPhrase, bool clearCache)
+        {
+            this.SaveService.UpdateEncrypted(key, value, passPhrase, clearCache);
+
+            if (clearCache)
+            {
+                this.Settings = this.LoadSettingsCallback();
```

Comment:
Would it make more sense to set this to `null`, and then it'll only re-query if someone needs to read again?